pdns/iputils.hh

```diff
@@ -261,6 +261,16 @@ union ComboAddress {
     return ret;
   }
 
+  bool isValid() const
+  {
+    char host[1024];
+    int retval = 0;
+    if(sin4.sin_family && !(retval = getnameinfo(reinterpret_cast<const struct sockaddr*>(this), getSocklen(), host, sizeof(host),0, 0, NI_NUMERICHOST)))
```

Comment:
what exactly does this check?

Reply:
We ran into a situation with the logging where if there was no edns info passed along, it still tried to print out the edns info, and it came out as 'Invalid ' because the toString method could not render it.  I wanted a way to check if there was something in the scope and subnet at all, so I used the same check that toString did, but had it let me know if there was or was not something printable there.  If there isn't, then we don't try to log it out.